ext/src/http/client/curl/http_client_curl.cc

```diff
@@ -57,11 +60,85 @@ nostd::shared_ptr<HttpCurlGlobalInitializer> HttpCurlGlobalInitializer::GetInsta
   return shared_initializer;
 }
 
+#ifdef ENABLE_OTLP_COMPRESSION_PREVIEW
+// Original source:
+// https://stackoverflow.com/questions/12398377/is-it-possible-to-have-zlib-read-from-and-write-to-the-same-memory-buffer/12412863#12412863
+int deflateInPlace(z_stream *strm, unsigned char *buf, uint32_t len, uint32_t *max)
+{
+  // must be large enough to hold zlib or gzip header (if any) and one more byte -- 11 works for the
+  // worst case here, but if gzip encoding is used and a deflateSetHeader() call is inserted in this
+  // code after the deflateReset(), then the 11 needs to be increased to accommodate the resulting
+  // gzip header size plus one
+  std::array<unsigned char, 11> temp{};
+
+  // kick start the process with a temporary output buffer -- this allows deflate to consume a large
+  // chunk of input data in order to make room for output data there
+  strm->next_in  = buf;
+  strm->avail_in = len;
+  if (*max < len)
+  {
+    *max = len;
+  }
+  strm->next_out  = temp.data();
+  strm->avail_out = std::min(static_cast<decltype(z_stream::avail_out)>(temp.size()), *max);
```

Comment:
```cpp
#ifdef min
#undef min
#endif
```

Should we add `#undef min`? Windows SDK may define `min` and will cause conflict sometimes.

Reply:
Given that min is a macro on windows, a work around is `(std::min)(a, b)`, like we do currently for max.

Reply:
Thanks @marcalff, I used this approach. @owent let me know if this is acceptable.

Reply:
Yes, but the variable `max` may still have problem. Or we can rename it?

Reply:
max is a macro in windows, so only the syntax max(...), with parenthesis, will cause breaks.

I think this should be fine.

Reply:
I went ahead and renamed it to avoid any surprises